lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -271,9 +269,39 @@ def update_mongos_hosts(self):
         for relation in self.charm.model.relations[self.relation_name]:
             self._update_relation_data(relation.id, {HOSTS_KEY: json.dumps(self.charm._unit_ips)})
 
-    def get_config_server_status(self):
-        """TODO: Implement this function in a separate PR."""
-        return None
+    def get_config_server_status(self) -> StatusBase:
+        """Returns the current status of the config-server."""
+        if not self.charm.is_role(Config.Role.CONFIG_SERVER):
+            logger.info("skipping status check, charm is not running as a shard")
+            return None
+
+        if not self.charm.db_initialised:
+            logger.info("No status for shard to report, waiting for db to be initialised.")
+            return None
+
+        if self.model.relations[LEGACY_REL_NAME]:
+            return BlockedStatus(f"relation {LEGACY_REL_NAME} to shard not supported.")
+
+        if self.model.relations[REL_NAME]:
+            return BlockedStatus(f"relation {REL_NAME} to shard not supported.")
+
+        if not self.is_mongos_running():
+            return BlockedStatus("Internal mongos is not running.")
+
+        shard_draining = self.get_draining_shards()
+        if shard_draining:
+            shard_draining = ",".join(shard_draining)
+            return MaintenanceStatus(f"Draining shard {shard_draining}")
+
+        if not self.model.relations[self.relation_name]:
+            return BlockedStatus("missing relation to shard(s)")
+
+        unreachable_shards = self.get_unreachable_shards()
+        if unreachable_shards:
+            unreachable_shards = "".join(unreachable_shards)
```

Comment:
on line 293 `join` uses "," while here it is not. I think it is better to use comma here as well